src/TeamRegistry.sol

```diff
@@ -0,0 +1,192 @@
+    error IsTeamLeader(uint32 teamId, address leader);
+
+    /// @notice Emitted when the user does not have a pending invite.
+    /// @param  teamId The ID of the team.
+    /// @param  member The member of the team.
+    error NoPendingInvite(uint32 teamId, address member);
+
+    /// @notice Emitted when the player is not a team member.
+    /// @param  teamId The ID of the team.
+    /// @param  player The player.
+    error NotInTeam(uint32 teamId, address player);
+
+    /// @notice Emitted when the caller is not the team leader.
+    /// @param  teamId The ID of the team.
+    /// @param  player The member of the team.
+    error NotTeamLeader(uint32 teamId, address player);
+
+    // -------------------------------------------------------------------------
+    // Events
+    // -------------------------------------------------------------------------
+
+    /// @notice Emitted when a new team is created.
+    /// @param  teamId The ID of the team.
+    /// @param  leader The leader of the team.
+    event TeamCreated(uint32 teamId, address leader);
+
+    /// @notice Emitted when a team invite has been accepted.
+    /// @param  teamId The ID of the team.
+    /// @param  member The member of the team that accepted the invite.
+    event TeamInviteAccepted(uint32 teamId, address member);
+
+    /// @notice Emitted when team leadership is transferred.
+    /// @param  oldLeader The old leader of the team.
+    /// @param  newLeader The new leader of the team.
+    event TeamLeadershipTransferred(uint32 teamId, address oldLeader, address newLeader);
+
+    // -------------------------------------------------------------------------
+    // Storage
+    // -------------------------------------------------------------------------
+
+    /// @notice Mapping for keeping track of which members are part of teams.
+    /// @dev Returns a uint8 that determines status of the team member.
+    ///      `0`: Neither in team nor invited
+    ///      `1`: Invited to team but not yet accepted
+    ///      `2`: Accepted invite and is team member
+    ///      `3`: Is team leader
+    mapping(uint32 => mapping(address => uint8)) public teamMemberStatus;
+
+    /// @notice Returns boolean if the address is a member of a team.
+    mapping(address => bool) public isTeamMember;
+
+    /// @notice Team ID counter. Starts at 1.
+    uint16 private teamId;
+
+    // -------------------------------------------------------------------------
+    // Functions
+    // -------------------------------------------------------------------------
+
+    /// @notice Create a team with a list of members.
+    /// @param  _members The team members to create the team with.
+    /// @dev    Reverts if `msg.sender` is already in a team.
+    function createTeam(address[] calldata _members) external {
+        // team leaders cannot create multiple teams
+        if (isTeamMember[msg.sender] == true) revert AlreadyInTeam(msg.sender);
```

Comment:
agree w/ the comment: leaders shouldn't be able to create teams, but I think team members should be able to create teams (upon creating, they'd immediately leave whatever team they used to be in and become the leader). maybe we can just add `_currentTeamId` as a parameter to make this part easy